common/openapi.yml

```diff
@@ -50,8 +50,105 @@ paths:
                 $ref: '#/components/schemas/DatasetsListResponse'
         '500':
           $ref: '#/components/responses/ServerError'
+    post:
+      tags: [ datasets ]
+      description: Create a new dataset
+      operationId: createDataset
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateDatasetRequest'
+        required: true
+      responses:
```

Comment:
Do we also want to include 401 errors here?

Reply:
Good point, I'll add 401 to these endpoints. I've added a note to [DC-279](https://broadworkbench.atlassian.net/browse/DC-279) to add tests for 401, 403, and other expected failure cases.

Reply:
After looking into this more, I think it's done this way because the 401 is returned by the OIDC gateway proxy, not by the application code. So the catalog service doesn't generate it or have control over its content (e.g., the result won't be a json payload of type `ErrorReport`). I'm going to follow the convention of other services and leave it out for now, but it would be easy to add it later if we decide that documenting a the 401 here in swagger would be helpful.